File: github-analyzer-server/src/main/kotlin/bme/softwarearchitectures/githubanalyzerserver/service/DoubleRepositoryService.kt
package bme.softwarearchitectures.githubanalyzerserver.service

import bme.softwarearchitectures.githubanalyzerserver.dto.compare.DeveloperCompareResponse
import bme.softwarearchitectures.githubanalyzerserver.dto.compare.DevelopmentCompareResponse
import bme.softwarearchitectures.githubanalyzerserver.dto.compare.DoubleRepositoryRequest

interface DoubleRepositoryService {

    fun analyze(request: DoubleRepositoryRequest)

    fun getDevelopmentCompareResult(request: DoubleRepositoryRequest): DevelopmentCompareResponse?

    fun getDeveloperCompareResult(request: DoubleRepositoryRequest): DeveloperCompareResponse?
}

File: github-analyzer-app/src/app/service/double-repository.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {environment} from '../../environments/environment';
import {DoubleRepositoryRequest} from '../model/double-repository-request.model';
import {DevelopmentCompareResponse} from '../model/development-compare-response.model';
import {DeveloperCompareResponse} from '../model/developer-compare-response.model';
import {RepositoryService} from './repository.service';

export const DOUBLE_REPOSITORY_API = environment.apiUrl + '/api/repository/double';

@Injectable()
export class DoubleRepositoryService extends RepositoryService {

  request: DoubleRepositoryRequest = new DoubleRepositoryRequest();

  constructor(httpClient: HttpClient) {
    super(httpClient);
  }

  getAPI(): string {
    return DOUBLE_REPOSITORY_API;
  }

  getRequest() {
    return this.request;
  }

  getDevelopmentCompareResult() {
    const url = DOUBLE_REPOSITORY_API + '/development';
    return this.httpClient.post<DevelopmentCompareResponse>(url, this.request);
  }

  getDeveloperCompareResult() {
    const url = DOUBLE_REPOSITORY_API + '/developer';
    return this.httpClient.post<DeveloperCompareResponse>(url, this.request);
  }
}
